client/src/GameState.ts

```diff
@@ -298,10 +308,12 @@ export class GameState {
                         const newMonomino = this.board[row][newCol];
                         if (newMonomino) {
                             newMonomino.position[1] = newCol;
+                            needRedraw.push(newMonomino);
                         }
                     }
                 }
             });
+        return needRedraw;
```

Comment:
we only really need to draw monominoes that has fallen due to a line clear, therefore an array `needRedraw`. Ideally the update function in scene just takes this array and redraw directly but due to we also get incoming events without scene's knowledge, I used a queue to cache the monominoes and each frame the scene will drain the queue and render them. Let me know what you think of this workaround. @swebra

Reply:
Yeah, totally makes sense to me.